mailet/api/src/main/java/org/apache/mailet/AttributeUtils.java

```diff
@@ -0,0 +1,52 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet;
+
+import java.util.Optional;
+
+/** 
+ * Attribute
+ * 
+ * @since Mailet API v3.2
+ */
+public class AttributeUtils {
+    public static <T> Optional<T> getValueAndCastFromMail(Mail mail, AttributeName name, Class<T> type) {
```

Comment:
Why can't we do that as part of the mail API?

I mean

`mail.getAttribute(name, clazz);`

Is more readable than:

`AttributeUtils.getValueAndCastFromMail(mail, name, clazz);`

Maybe using a default method can enable use to easily leverage current API without being invasive on the implementations...